pdc/apps/auth/middleware.py

```diff
@@ -48,9 +49,12 @@ def process_request(self, request):
                 return
         # We are seeing this user for the first time in this session, attempt
         # to authenticate the user.
-        user = auth.authenticate(remote_user=username)
+        user = auth.authenticate(remote_user=username, request=request)
         if user:
             # User is valid.  Set request.user and persist user in the session
             # by logging the user in.
             request.user = user
-            auth.login(request, user)
+            request.session['auth_backend'] = user.backend
+            backend = load_backend(user.backend)
+            if not hasattr(backend, 'save_login') or not backend.save_login:
```

Comment:
This line is here because it will allow mod_auth_mellon to control the login state.
The advantage of this is that is that we get Single Logout for free.